### DISCUSSION TITLE
It needs serious UX and UI work

### DISCUSSION THREAD

**Discussion by fdevin** [woocommerce/woocommerce#50421] [Checkout Flow]
Trying to add a simple field. Absolutely nothing works, not even a React injection. 

This should be a simple drag and drop form with the ability to order fields like most plugins that uses forms. I didn't use WooCommerce for a while and got very unpleasantly surprised when I find that now it's completely limited and everything that works in absolutely every plugin or WP core doesn't work in WC. A real step back.

**Comment by senadir** [comment]
Have you tried our additional fields API?
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md
